src/main/ruby/truffleruby/core/truffle/string_operations.rb

```diff
@@ -411,5 +411,16 @@ def self.assign_regexp(string, index, count, replacement)
 
       Primitive.string_splice(string, replacement, bi, bs, enc)
     end
+
+    def self.on_codepoint_boundary?(string, byte_pos)
```

Comment:
We already have a way to check this, IsCharacterHeadNode, could you expose via a Primitive in StringNodes and use it instead?

Reply:
Sure. I did it in the `String#bytesplice` branch, I'll wait till that is in a good shape.